pkg/file/splitter/splitter_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package splitter_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/file/splitter"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+	mockbytes "gitlab.com/nolash/go-mockbytes"
+)
+
+// TestSplitIncomplete tests that the Split method returns an error if
+// the amounts of bytes written does not match the data length passed to the method.
+func TestSplitIncomplete(t *testing.T) {
+	testData := make([]byte, 42)
+	store := mock.NewStorer()
+	s := splitter.NewSimpleSplitter(store)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	testDataReader := file.NewSimpleReadCloser(testData)
+	_, err := s.Split(ctx, testDataReader, 41)
+	if err == nil {
+		t.Fatalf("expected error on EOF before full length write")
+	}
+}
+
+// TestSplitSingleChunk hashes one single chunk and verifies
+// that that corresponding chunk exist in the store afterwards.
+func TestSplitSingleChunk(t *testing.T) {
+
+	// edge case selected from internal/job_test.go
+	g := mockbytes.New(0, mockbytes.MockTypeStandard).WithModulus(255)
+	testData, err := g.SequentialBytes(swarm.ChunkSize)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	store := mock.NewStorer()
+	s := splitter.NewSimpleSplitter(store)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
There are no explicit parallel work being done in these tests, so cancel is not canceling any goroutine. I think that it is ok just to have context.Background() instead in all tests in this file.